index.js

```diff
@@ -1,11 +1,14 @@
 import { json } from 'http-responders'
 import { migrate } from './lib/migrate.js'
+import getRawBody from 'raw-body'
+import assert from 'http-assert'
 
 export const createHandler = async (client) => {
   await migrate(client)
 
   const handler = async (req, res) => {
-    if (req.url === '/retrievals' && req.method === 'POST') {
+    const segs = req.url.split('/').filter(Boolean)
+    if (segs[0] === 'retrievals' && req.method === 'POST') {
```

Comment:
This is a subtle behaviour change. With your change in place, we start handling all paths starting with `/retrievals`, e.g. `/retrievals/foobar`.

A possible fix:

```suggestion
    if (segs[0] === 'retrievals' && segs.length == 1 && req.method === 'POST') {
```

I wonder how far we can get with this bare-bone DYI approach before having to switch to a proper web framework like Fastify anyways. 🤔

Reply:
That changed yeah, but does it matter?

> I wonder how far we can get with this bare-bone DYI approach before having to switch to a proper web framework like Fastify anyways. 🤔

Let's go there when we need to. We can also try some more lightweight approaches first, like https://github.com/pillarjs/path-to-regexp or writing regexps directly.

Reply:
At this stage I would like to keep complexity low and movability high, we can prevent things like `POST /retrievals/mistake` further down the road.

Reply:
For reference, I built the electron update server this way too https://github.com/electron/update.electronjs.org/blob/main/src/updates.js#L62-L84, and that hasn't needed changes related to the routing layer for 5 years